scripts/megastock/write_databricks_to_bigquery_03.py

```diff
@@ -18,52 +25,116 @@
 
 # COMMAND ----------
 
+from itertools import chain
+
 from google.cloud import bigquery
+import pyspark.sql.functions as F
+from pyspark.sql.functions import udf
+from pyspark.sql.types import IntegerType
 
 # COMMAND ----------
 
 N_SAMPLE_TAG = dbutils.widgets.get("n_sample_tag")
 
 # COMMAND ----------
 
-bq_metadata_write_path = 'cube-machine-learning.ds_api_datasets.megastock_metadata'
+# Initialize a BigQuery client
+client = bigquery.Client()
+
+# set up paths to write to 
+bq_project = "cube-machine-learning"
+bq_dataset = "ds_api_datasets"
+bq_megastock_table = 'megastock_combined_baseline'
+bq_write_path = f"{bq_project}.{bq_dataset}.{bq_megastock_table}"
+
+# COMMAND ----------
+
+# read in data
 building_metadata = spark.table(f'ml.megastock.building_metadata_{N_SAMPLE_TAG}')
+building_features = spark.table(f'ml.megastock.building_features_{N_SAMPLE_TAG}')
+
+# COMMAND ----------
+
+# get distinct climate zones
+climate_zone_list = list(map(lambda row: row['ashrae_iecc_climate_zone_2004'], building_metadata.select('ashrae_iecc_climate_zone_2004').distinct().collect()))
 
+# Mapping dictionary
+climate_zone_mapping = {cz: i for i, cz in (enumerate(sorted(climate_zone_list), 1))}
```

Comment:
I think it would be safer to hardcode this list (like you did in the `dohyo` PR), to guarantee that the two stay in sync and don't depend on the contents of the building metadata table. And then maybe put it in a shared location, like dml-utils?

Reply:
Yeah I debated this and went back and forth. I'll hard code it. I have [another ticket ](https://www.notion.so/rewiringamerica/Align-Feature-Upgrade-code-in-Sumo-and-Dohyo-1514c9af1a1c800caccae6a1ae4647dc?pvs=4#1514c9af1a1c80e2a2e9d80f9a835c23)for a whole bunch of things to use shared code for dohyo and sumo by using dmlutils and I'll add that to the list. I'm not gonna do it as part of this PR.